frontend/app/src/components/StreamlitDialog/SettingsDialog.test.tsx

```diff
@@ -53,49 +54,53 @@ const getProps = (extend?: Partial<Props>): Props => ({
   developerMode: true,
   animateModal: true,
   openThemeCreator: jest.fn(),
-  metricsMgr: new SegmentMetricsManager(mockSessionInfo()),
+  metricsMgr: new SegmentMetricsManager(
+    // @ts-expect-error The mock seems to have a mismatched internal type to what's expected.
+    mockSessionInfo()
+  ),
   ...extend,
 })
 
 describe("SettingsDialog", () => {
-  // Note: You may need to wrap the component in a context provider to pass the context if necessary.
-
   it("renders without crashing", () => {
     const availableThemes = [lightTheme, darkTheme]
     const props = getProps()
     const context = getContext({ availableThemes })
 
     customRenderLibContext(<SettingsDialog {...props} />, context)
 
-    expect(screen.getByText("Settings")).toBeInTheDocument()
+    expect(screen.getByText("Settings")).toBeVisible()
   })
 
-  it("should render run on save checkbox", () => {
+  it("should render run on save checkbox", async () => {
+    const user = userEvent.setup()
     const props = getProps({
       allowRunOnSave: true,
     })
     const context = getContext()
     customRenderLibContext(<SettingsDialog {...props} />, context)
-    expect(screen.getByText("Run on save")).toBeInTheDocument()
 
-    screen.getByText("Run on save").click()
+    await user.click(screen.getByText("Run on save"))
```

Comment:
Curious question: what is the benefit of `user.click()` vs `screen.getByText("Run on save").click()`? From reading the docs, it sounds like that `user-events` is a more realistic simulation of user events. It might be good to start a list of tips & tricks in our wiki related to writing RTL tests.

Reply:
That's correct, `user-event` is more realistic to how a user interacts with the application. Generally speaking I prefer to lean on automated enforcement rather than documentation, so in that spirit, I have a follow-up PR that enforces `user-event` via eslint [here](https://github.com/streamlit/streamlit/pull/9747)